psci/PSCi.hs

```diff
@@ -380,17 +381,50 @@ printModuleSignatures :: P.ModuleName -> P.Environment -> PSCI ()
 printModuleSignatures moduleName env =
   PSCI $ let namesEnv = P.names env
              moduleNamesIdent = (filter ((== moduleName) . fst) . M.keys) namesEnv
+             typesEnv = P.types env
+             typeSynonymsEnv = P.typeSynonyms env
+             moduleTypes = (filter (\(P.Qualified maybeName _) -> maybeName == Just moduleName) . M.keys) typesEnv
+             dataConstructorsEnv = P.dataConstructors env
+             moduleDataConstructors = (filter (\(P.Qualified maybeName _) -> maybeName == Just moduleName) . M.keys) dataConstructorsEnv
              in case moduleNamesIdent of
```

Comment:
Why not factor this into a function which produces a list, and then a function which renders it? Then you can write this check using `null`.